react/utilities/index.ts

```diff
@@ -0,0 +1,12 @@
+import { useState, useCallback } from 'react'
+
+export const useDisclosure = (open = false) => {
+  const [isOpen, setIsOpen] = useState(open)
+
+  return {
+    isOpen,
+    onOpen: useCallback(() => setIsOpen(true), []),
```

Comment:
Doesn't this usage break the first rule of hooks? 🤔.

Reply:
Fixed it